libs/designer-ui/src/lib/dynamicallyaddedparameter/index.tsx

```diff
@@ -0,0 +1,138 @@
+
+  const intl = useIntl();
+  const menuButtonTitle = intl.formatMessage({
+    defaultMessage: 'Menu',
+    description: 'Open dynamically added parameter options menu',
+  });
+
+  const renderMenuButton = (): JSX.Element => {
+    const deleteText = intl.formatMessage({
+      defaultMessage: 'Delete',
+      description: 'Delete dynamic parameter corresponding to this row',
+    });
+
+    const menuProps: IContextualMenuProps = {
+      shouldFocusOnMount: true,
+      items: [
+        {
+          iconProps: { iconName: 'Delete' },
+          key: guid(),
+          text: deleteText,
+          onClick: (ev?: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>, _item?: IContextualMenuItem) => {
+            ev?.preventDefault();
+            const { onDelete } = props;
+            if (onDelete) onDelete(props.schemaKey);
+            return true;
+          },
+        },
+        { key: 'divider_1', itemType: ContextualMenuItemType.Divider },
+      ],
+      gapSpace: 2,
+      directionalHint: DirectionalHint.leftBottomEdge,
+    };
+
+    return (
+      <TooltipHost content={menuButtonTitle}>
+        <IconButton
+          className="msla-button msla-card-header-menu-button"
+          iconProps={{ iconName: 'CollapseMenu' }}
+          title={menuButtonTitle}
+          aria-label={menuButtonTitle}
+          ref={menuRef}
+          menuProps={menuProps}
+        />
+      </TooltipHost>
+    );
+  };
+
+  const onTitleChange = (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newPropertyValue?: string) => {
+    e.preventDefault();
+    const { onChange } = props;
+    if (onChange) onChange(props.schemaKey, /* propertyName */ 'title', newPropertyValue);
+  };
+
+  const onDescriptionChange = (e: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>, newPropertyValue?: string) => {
+    e.preventDefault();
+    const { onChange } = props;
+    if (onChange) onChange(props.schemaKey, /* propertyName */ 'description', newPropertyValue);
+  };
+
+  const renderDynamicParameterContainer = (): JSX.Element => {
+    const iconStyle = {
+      background: `url('${props.icon}') no-repeat center`,
+      backgroundSize: 'contain',
+    };
+
+    return (
+      <div>
+        <div className="msla-dynamic-added-param-header">
+          <div className="msla-dynamic-added-param-icon" style={iconStyle}></div>
+          <div className="msla-dynamic-added-param-inputs-container">
+            <TextField value={props.properties.title} onChange={onTitleChange} />
+            <TextField value={props.properties.description} onChange={onDescriptionChange} />
+          </div>
+          <div className="msla-dynamic-add-param-menu-container">{renderMenuButton()}</div>
+        </div>
+      </div>
+    );
+  };
+
+  return (
+    <div className="msla-dynamic-added-param-container">
+      {renderDynamicParameterContainer()}
+      <div className="msla-dynamic-added-param-bottom-divider" />
+    </div>
+  );
+};
```

Comment:
Error when loading flow in v1 designer below. This is because v1 expects certain properties to be structured in a way that is missing in v3.

TODO: figure out importance of this
![image](https://user-images.githubusercontent.com/10837129/233423782-3dfeb6e5-a64d-4330-858f-bb9bc399d102.png)

Reply:
This should not be blocking signoff for this PR. I will take as an action item to talk about it